internal/pkg/deploy/cloudformation/pipeline.go

```diff
@@ -25,8 +37,33 @@ func (p *pipelineStackConfig) StackName() string {
 }
 
 func (p *pipelineStackConfig) Template() (string, error) {
-	// TODO: Render the template
-	return "", nil
+	content, err := templates.Box().FindString(pipelineCfnTemplatePath)
+	if err != nil {
+		return "", &ErrTemplateNotFound{templateLocation: pipelineCfnTemplatePath, parentErr: err}
+	}
+	
+	tpl, err := template.New("pipelineCfn").Parse(removeComments(content))
+	if err != nil {
+		return "", fmt.Errorf("parse CloudFormation template for project %s, pipeline %s, error: %w",
+		p.ProjectName, p.Name, err)
+	}
+	var buf bytes.Buffer
+	if err := tpl.Execute(&buf, p); err != nil {
+		return "", fmt.Errorf("execute CloudFormation template for project %s, pipeline %s, error: %w",
+		p.ProjectName, p.Name, err)
+	}
+	return buf.String(), nil
+}
+
+func removeComments(tmpl string) string {
```

Comment:
Why remove the comments? They shouldn't hurt to leave them in, right?

Reply:
If customers are going to read them then it makes a lot more sense to leave them in :). I was hoping to reduce the size of CFN template because I've run into the body size limit previously when sending template to Cloudformation: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cloudformation-limits.html

what do you think?

Reply:
Ah, I see. Why don't we get rid of the comments in the template. There's no need having them in there if we're not going to deploy them :D

Reply:
It's mostly for myself because there's too much information to keep in the "cache" sometimes.

Reply:
I'd lean towards keeping the comments and making them less verbose, the current template is 11820 bytes (unrendered) even after adding the stages I don't think it'll grow to the cap.

I'm a fan of not writing code :P as it's another place where we could introduce bugs. So let's either remove all the comments or keep them in and not filter them with a regex.

Reply:
The rendered template (minus all the comments) is about 11143 bytes, slightly more than 1/5 of the limit. Since this template is quite large and confusing. I would love to keep the verbose comments. David suggested to remove comments by yaml unserialize/serialize so I will probably do that.

Reply:
Update: Decided to removed some comments and keep the rest in the final rendered CFN template for now.